app/javascript/mastodon/components/hover_card.tsx

```diff
@@ -0,0 +1,205 @@
+const enterDelay = 250;
+const leaveDelay = 300;
+
+export const HoverCard: React.FC = () => {
+  const [open, setOpen] = useState(false);
+  const [accountId, setAccountId] = useState<string | undefined>();
+  const [anchor, setAnchor] = useState<HTMLElement | null>(null);
+  const cardRef = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+  const account = useAppSelector((state) =>
+    accountId ? state.accounts.get(accountId) : undefined,
+  );
+  const leaveTimerRef = useRef<ReturnType<typeof setTimeout>>();
+  const enterTimerRef = useRef<ReturnType<typeof setTimeout>>();
+
+  const handleAnchorMouseEnter = useCallback(
+    (e: MouseEvent) => {
+      const { target } = e;
+
+      if (
+        target instanceof HTMLElement &&
+        target.matches('[data-hover-card]')
+      ) {
+        clearTimeout(leaveTimerRef.current);
+        clearTimeout(enterTimerRef.current);
+
+        enterTimerRef.current = setTimeout(() => {
+          target.setAttribute('aria-describedby', 'hover-card');
+          setAnchor(target);
+          setOpen(true);
+          setAccountId(target.getAttribute('data-hover-card') ?? undefined);
+        }, enterDelay);
+      }
+
+      if (target === cardRef.current?.parentNode) {
+        clearTimeout(leaveTimerRef.current);
+      }
+    },
+    [setOpen, setAccountId, setAnchor],
+  );
+
+  const handleAnchorMouseLeave = useCallback(
+    (e: MouseEvent) => {
+      if (e.target === anchor || e.target === cardRef.current?.parentNode) {
+        clearTimeout(leaveTimerRef.current);
+        clearTimeout(enterTimerRef.current);
+
+        leaveTimerRef.current = setTimeout(() => {
+          anchor?.removeAttribute('aria-describedby');
+          setOpen(false);
+          setAnchor(null);
+        }, leaveDelay);
+      }
+    },
+    [setOpen, setAnchor, anchor],
+  );
+
+  const handleFocus = useCallback(
+    (e: FocusEvent) => {
+      const { target } = e;
+
+      if (
+        target instanceof HTMLElement &&
+        target.matches('[data-hover-card]')
+      ) {
+        target.setAttribute('aria-describedby', 'hover-card');
+        setAnchor(target);
+        setOpen(true);
+        setAccountId(target.getAttribute('data-hover-card') ?? undefined);
+      }
+    },
+    [setAnchor, setOpen, setAccountId],
+  );
+
+  const handleBlur = useCallback(
+    (e: FocusEvent) => {
+      const { target } = e;
+
+      if (
+        target instanceof HTMLElement &&
+        target.matches('[data-hover-card]')
+      ) {
+        target.removeAttribute('aria-describedby');
+        setOpen(false);
+        setAnchor(null);
+      }
+    },
+    [setOpen, setAnchor],
+  );
+
+  const handleClose = useCallback(() => {
+    clearTimeout(leaveTimerRef.current);
+    clearTimeout(enterTimerRef.current);
+    setOpen(false);
+    setAnchor(null);
+  }, [setOpen, setAnchor]);
+
+  useEffect(() => {
+    document.body.addEventListener('mouseenter', handleAnchorMouseEnter, {
```

Comment:
I wonder if we should rather use `<div onMouseEnter=… …>{children}</div>` to attach the elements only to this specific part of the DOM.

In some places (admin/moderation) we render multiple React Roots, and attaching to `document.body` for each of those (if we choose to use this component in those views) will cause issues, as each callback will be called on each hover